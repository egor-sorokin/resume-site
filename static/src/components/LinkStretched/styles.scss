@import '../../styles/base/_vars.scss';

.link-stretched {
  letter-spacing: 3px;

  transform: translateX(-40px);
  transition: transform $defaultTiming $defaultEasing,
  letter-spacing $defaultTiming $defaultEasing;
}

.link-stretched:hover {
  letter-spacing: 5px;

  transform: translateX(-34px);
}
