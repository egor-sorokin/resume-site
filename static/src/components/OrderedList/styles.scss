@import '../../styles/base/_vars.scss';

.orderedlist {
  &__list {
    list-style: none;
    counter-reset: item - 1;
  }

  &__item {
    counter-increment: item;
    position: relative;
    display: flex;
    flex-direction: row;


    &:before {
      content: counter(item, decimal-leading-zero);
      margin-left: 10px;

      min-width: 20px;
    }
    &:after {
      content: '';
      display: block;

      position: absolute;
      top: 14px;
      left: 0;

      width: 5px;
      height: 1px;

      background-color: $mercuryLight;
    }
    &-dash {
      margin-top: 13px;
      margin-right: 10px;

      width: 15px;
      height: 1px;

      background-color: $mercuryLight;
    }
  }
  &__text {
    color: $mercuryLight;
  }
  &__title {
    font-family: $secondaryFont;
    font-size: 12px;
    letter-spacing: 3px;
    text-align: left;
    text-transform: uppercase;
    font-weight: 700;
  }
}
