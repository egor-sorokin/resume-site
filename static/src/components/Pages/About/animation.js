import {Power2} from 'gsap';


export const attachToggleAnimation = (targetTween) => {
  targetTween
    .to('.about', .8,
      {x: 0, ease: Power2.easeInOut}
    )
    .fromTo('.about .about__title', .4,
      {x: "30%", opacity: 0, ease: Power2.easeOut},
      {x: "0%", opacity: 1, ease: Power2.easeOut}
    )
    .staggerFromTo('.about .summary__item', .6,
      {x: "20%", opacity: 0, ease: Power2.easeOut},
      {x: "0%", opacity: 1, ease: Power2.easeOut},
      0.1, "-=0.1"
    )
    .fromTo('.about .orderedlist__title', .6,
      {opacity: 0, ease: Power2.easeOut},
      {opacity: 1, ease: Power2.easeOut},
      "-=0.1"
    )
    .staggerFromTo('.about .orderedlist__item', .3,
      {x: 100, opacity: 0, ease: Power2.easeOut},
      {x: 0, opacity: 1, ease: Power2.easeOut},
      0.15, "-=0.5"
    )
    .fromTo('.about .logo', .4,
      {opacity: 0, ease: Power2.easeOut},
      {opacity: 1, ease: Power2.easeOut},
      "-=0.1"
    )
    .fromTo('.about .logo__circle-big', .5,
      {rotation: -180, transformOrigin: '75% 19%', ease: Power2.easeOut},
      {rotation: 0, transformOrigin: '75% 19%', ease: Power2.easeOut},
      "-=0.2"
    )
    .fromTo('.about .logo__circle-medium', .5,
      {rotation: 180, transformOrigin: '60% 70%', ease: Power2.easeOut},
      {rotation: 0, transformOrigin: '60% 70%', ease: Power2.easeOut},
      "-=0.5"
    )
    .fromTo('.about .button-close', .3,
      {x: 500, opacity: 0, ease: Power2.easeOut},
      {x: 0, opacity: 1, ease: Power2.easeOut},
      "-=0.3"
    )
    .reversed(true);
};
