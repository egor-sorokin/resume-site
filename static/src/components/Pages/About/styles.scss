@import '../../../styles/base/_vars.scss';

.about {
  display: flex;
  flex-direction: row;

  background-color: $dune;
  color: $mercuryLight;

  width: 100vw;
  height: 100vh;

  &__item {
    display: flex;
    flex-direction: column;

    position: relative;

    &-footer {
      margin-top: 10vh;
    }
    &--left {
      display: flex;
      flex-direction: column;
      flex-basis: 20vw;
    }
    &--middle {
      padding-left: 8vw;
      flex-basis: 52vw;
    }
    &--right {
      flex-basis: 28vw;
      padding-right: 5vw;
      padding-left: 8vw;

      align-self: center;
    }

    &--right,
    &--middle {
      padding-top: 10vh;
      padding-bottom: 10vh;
    }
  }

  &__title {
    font-size: 2.2em;
    text-transform: uppercase;
    text-align: center;
    letter-spacing: 5px;
  }
}

.about {
  .line {
    position: absolute;
    left: 50%;

    height: 40vh;
    width: 1px;
    background-color: $mercuryLight;

    transform: translateX(-50%);

    &--bottom {
      bottom: 0;
    }
    &--top {
      top: 0;
    }
  }
}

.about {
  .logo {
    position: absolute;
    left: 50%;
    top: 50%;

    transform: translateY(-50%) translateX(-50%);
  }
}

