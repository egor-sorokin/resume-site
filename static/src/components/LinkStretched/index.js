import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = props => {
  const {text, url, target, cssClasses, onClick} = props;

  const handleClick = (e) => {
    if (!url) {
      e.preventDefault();
      onClick();
    } else {
      onClick(e.currentTarget.dataset.anchorId.substring(1), true);
    }
  };

  return (
    <a
      href={url}
      data-anchor-id={url}
      target={target}
      onClick={handleClick}
      className={'link-stretched ' + cssClasses}
    >{text}</a>
  )
};


LinkStretched.propTypes = {
  onClick: PropTypes.func,
  cssClasses: PropTypes.string,
  text: PropTypes.string,
  url: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  cssClasses: '',
  text: '',
  url: '',
  target: '',
  onClick: () => {
  }
};


export default LinkStretched;