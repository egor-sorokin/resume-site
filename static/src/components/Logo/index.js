import React from 'react';
import './styles.scss';


const Logo = () => (
  <div className="logo">
    <svg className="logo__circles" height="87" viewBox="0 0 88 87" width="88">
      <filter id="a" height="107.2%" width="104.7%" x="-2.3%" y="-2.4%">
        <feOffset dy="2" in="SourceAlpha" result="shadowOffsetOuter1"/>
        <feColorMatrix in="shadowOffsetOuter1" result="shadowMatrixOuter1"
                       values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0"/>
        <feMerge>
          <feMergeNode in="shadowMatrixOuter1"/>
          <feMergeNode in="SourceGraphic"/>
        </feMerge>
      </filter>
      <g fill="none" fillRule="evenodd" filter="url(#a)" stroke="#ddd" transform="translate(1 1)">
        <ellipse cx="43" cy="41.5" fill="#424242" rx="43" ry="41.5"/>
        <ellipse cx="53.238" cy="27.553" fill="#2d2b28" rx="25.254" ry="24.832"/>
        <ellipse cx="61.429" cy="18.709" fill="#696969" rx="13.651" ry="13.266"/>
        <ellipse cx="57.675" cy="65.992" fill="#696969" rx="13.992" ry="13.607"/>
        <ellipse cx="61.429" cy="70.074" fill="#2d2b28" rx="8.19" ry="8.164"/>
        <ellipse cx="15.016" cy="39.799" fill="#696969" rx="15.016" ry="14.627"/>
        <ellipse cx="26.619" cy="39.799" rx="26.619" ry="25.512"/>
        <ellipse cx="8.19" cy="40.139" fill="#909090" rx="8.19" ry="8.164"/>
      </g>
    </svg>
  </div>
);

export default Logo;
