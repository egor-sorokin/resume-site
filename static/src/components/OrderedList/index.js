import React from 'react';
import './styles.scss';


const OrderedList = () => {
  const orderedListItems = ['LinkedIn', 'Xing', 'GitHub', 'Codepen'].map((item) => (
    <li className="orderedlist__item">
      <span className="orderedlist__text">{item}</span>
    </li>
  ));

  return (
    <div className="orderedlist">
      <h5 className="orderedlist__title">Attachments</h5>
      <ol className="orderedlist__list" start="1">
        {orderedListItems}
      </ol>
    </div>
  )
};


export default OrderedList;