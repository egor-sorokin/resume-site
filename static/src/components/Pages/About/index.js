import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {TimelineMax} from 'gsap';
import {attachToggleAnimation} from './animation';
import OrderedList from '../../OrderedList/index'
import Summary from '../../Summary/index'
import Logo from '../../Logo/index';
import LinkStretched from '../../LinkStretched/index';
import {LINK_MASKED, BUTTON_CLOSE} from '../../../constants/index';
import './styles.scss';


class About extends Component {
  constructor(props) {
    super(props);

    this.aboutTween = null;
  }


  componentDidMount() {
    if (!this.props.isFetching) {
      this.aboutTween = new TimelineMax();
      attachToggleAnimation(this.aboutTween);
    }
  }


  componentDidUpdate(prepProps) {
    if (this.props.isOpenedAbout !== prepProps.isOpenedAbout) {
      this.aboutTween.reversed(!this.props.isOpenedAbout);
      document.body.style.overflowY = (this.props.isOpenedAbout ? 'hidden' : 'initial');
    }
  }


  clickCloseButton = () => {
    this.props.toggleAboutSection();
  };


  render() {
    const {data} = this.props;
    const personData = data.personData || {};
    const socials = personData.socials || [];
    const cssClassesButton = "button-close__link font-s-12-secondary text-c-mercury-light";

    return (
      <div>
        <section className="about text-c-mercury-light">
          <div className="about__item about__item--left">
            <div className="line line--top"></div>
            <Logo></Logo>
            <div className="line line--bottom"></div>
          </div>
          <div className="about__item about__item--middle">
            <h1 className="about__title font-s-36"
            >
              {personData.first_name + ' ' + personData.last_name}
            </h1>
            <Summary
              summary={personData.summary}
            ></Summary>
          </div>
          <div className="about__item about__item--right">
            <OrderedList
              title={socials.title}
              items={socials.items}
              type={LINK_MASKED}
            ></OrderedList>
          </div>
          <div className="button-close">
            <LinkStretched
              onClick={this.clickCloseButton}
              cssClasses={cssClassesButton}
              text={BUTTON_CLOSE.text}
            ></LinkStretched>
          </div>
        </section>
      </div>
    );
  }
}


About.propTypes = {
  isFetching: PropTypes.bool,
  data: PropTypes.shape({}),
  isOpenedAbout: PropTypes.bool,
  toggleAboutSection: PropTypes.func
};

About.defaultProps = {
  data: {},
  isFetching: true,
  isOpenedAbout: false,
  toggleAboutSection: () => {
  }
};


export default About;
