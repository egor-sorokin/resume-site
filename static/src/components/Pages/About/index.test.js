// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import About from './index';
import Summary from '../../Summary/index';
import OrderedList from '../../OrderedList/index';
import Logo from '../../Logo/index';
import LinkStretched from '../../LinkStretched/index';


describe('<About />', () => {
  let component;
  let spy;

  const tweenline = {
    play: () => {
    },
    reversed: () => {
    }
  };
  const person = {
    id: 1,
    first_name: 'Bob',
    last_name: 'Cat',
    email: 'example@admin.com',
    summary: [
      {
        title: 'test title',
        text: 'test text'
      },
      {
        title: 'test title 2',
        text: 'test text 2'
      }
    ],
    socials: {
      title: 'test title',
      items: [
        {
          id: 1,
          text: 'test text 1',
          url: 'test.urlone'
        },
        {
          id: 2,
          text: 'test text 2',
          url: 'test.urltwo'
        }
      ]
    }
  };
  const isFetching = false;
  const isOpenedAbout = true;
  const toggleAboutSection = jest.fn(() => {
  });
  const onClick = jest.fn(() => {
    toggleAboutSection();
  });


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 4 component', () => {
      expect(component.children()).toHaveLength(4);
    });
  });


  describe('props tests', () => {
    it('renders data', () => {
      component = mount(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      expect('person' in component.props()).toBeTruthy();
      expect(component.find(<Summary />)).toBeTruthy();
      expect(component.find(<OrderedList />)).toBeTruthy();
      expect(component.find(<Logo />)).toBeTruthy();
      expect(component.find(<LinkStretched />)).toBeTruthy();
      expect(component.find('.about__title')).toBeTruthy();

      component.setProps({person: {}});
    });
  });


  describe('close button functionality tests', () => {
    it('clicks on the close button', () => {
      component = mount(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={!isOpenedAbout}
        />
      );
      component.setProps({toggleAboutSection: toggleAboutSection});

      let closeButton = component.find('.button-close .link-stretched').at(0);
      expect(closeButton.prop('onClick')).toHaveLength(1);

      closeButton.simulate('click');

      expect(toggleAboutSection).toHaveBeenCalledTimes(1);
    });
  });


  describe('lifecycle tests', () => {
    beforeEach(() => {
      component = '';
    });


    afterEach(() => {
      spy.mockClear();
    });


    it('calls play an animation', () => {
      spy = jest.spyOn(tweenline, 'play');
      // eslint-disable-next-line
      component = shallow(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      component.setState({aboutTween: tweenline});
      component.state().aboutTween.play();
      expect(spy).toHaveBeenCalledTimes(1);
    });


    it('calls reversed animation', () => {
      spy = jest.spyOn(tweenline, 'reversed');
      // eslint-disable-next-line
      component = shallow(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      component.setState({aboutTween: tweenline});
      component.state().aboutTween.reversed();

      expect(spy).toHaveBeenCalledTimes(1);
    });


    it('calls componentDidMount', () => {
      spy = jest.spyOn(About.prototype, 'componentDidMount');
      // eslint-disable-next-line
      component = shallow(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      expect(spy).toHaveBeenCalledTimes(1);
    });


    it('calls componentDidUpdate', () => {
      spy = jest.spyOn(About.prototype, 'componentDidUpdate');
      // eslint-disable-next-line
      component = mount(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      component.setProps({ person });
      expect(spy).toHaveBeenCalledTimes(1);
    });


    it('calls componentDidUpdate with updated props', () => {
      spy = jest.spyOn(About.prototype, 'componentDidUpdate');
      // eslint-disable-next-line
      component = shallow(
        <About
          person={person}
          isFetching={isFetching}
          isOpenedAbout={isOpenedAbout}
        />
      );
      component.setState({aboutTween: tweenline});
      component.setProps({isOpenedAbout: true});

      expect(spy).toHaveBeenCalledTimes(2);
    });
  });
});
