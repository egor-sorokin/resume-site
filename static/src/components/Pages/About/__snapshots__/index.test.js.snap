// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> base tests matches snapshot 1`] = `
<div>
  <section
    className="about text-c-mercury-light"
  >
    <div
      className="about__item about__item--left"
    >
      <div
        className="line line--top"
      />
      <Logo />
      <div
        className="line line--bottom"
      />
    </div>
    <div
      className="about__item about__item--middle"
    >
      <h1
        className="about__title font-s-36"
      >
        undefined undefined
      </h1>
      <Summary
        summary={Array []}
      />
    </div>
    <div
      className="about__item about__item--right"
    >
      <OrderedList
        items={Array []}
        title=""
        type="mask"
      />
    </div>
    <div
      className="button-close"
    >
      <LinkStretched
        cssClasses="button-close__link font-s-12-secondary text-c-mercury-light"
        onClick={[Function]}
        target=""
        text="Close"
        url=""
      />
    </div>
  </section>
</div>
`;
